src/UrlBuilder.php

```diff
@@ -24,19 +24,25 @@ class UrlBuilder
      */
     private $secure = false;
 
+    /**
+     * @var int
+     */
+    private $signatureSize;
+
     /**
      * UrlBuilder constructor.
      * @param string $baseUrl
      * @param string $key
      * @param string $salt
      * @throws Exception
      */
-    public function __construct(string $baseUrl, string $key = null, string $salt = null)
+    public function __construct(string $baseUrl, string $key = null, string $salt = null, $signatureSize = 0)
     {
         if ($key && $salt) {
             $this->key = pack("H*" , $key) ?: $this->throwException("Key expected to be hex-encoded string");
             $this->salt = pack("H*" , $salt) ?: $this->throwException("Salt expected to be hex-encoded string");
             $this->secure = true;
+            $this->signatureSize = $signatureSize;
```

Comment:
Maybe it would have been nice to check some constraints like the number being in between 0 and including 32:

```php
if ($signatureSize < 0 || $signatureSize > 32) {
  $this->throwException("...");
}
```

Reply:
I'll look into this, thank you! Just a side note: if you like/use this project, feel free to add a star to it ;-)